build/panic_reporter.go

```diff
@@ -0,0 +1,164 @@
+package build
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime/debug"
+	"runtime/pprof"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/icza/backscanner"
+	logging "github.com/ipfs/go-log/v2"
+)
+
+var (
+	panicLog           = logging.Logger("panic-reporter")
+	defaultJournalTail = 500
+)
+
+// PanicReportingPath is the name of the subdir created within the repoPath path
+// provided to GeneratePanicReport
+var PanicReportingPath = "lotus-panic-reports"
+
+// PanicReportJournalTail is the number of lines captured from the end of
+// the lotus journal to be included in the panic report.
+var PanicReportJournalTail = defaultJournalTail
+
+// GeneratePanicReport produces a timestamped dump of the application state
+// for inspection and debugging purposes. Call this function from any place
+// where a panic or severe error needs to be examined. `persistPath` is the
+// path where the reports should be saved. `repoPath` is the path where the
+// journal should be read from. `label` is an optional string to include
+// next to the report timestamp.
+func GeneratePanicReport(persistPath, repoPath, label string) {
+	// make sure we always dump the latest logs on the way out
+	// especially since we're probably panicking
+	defer panicLog.Sync() //nolint:errcheck
+
+	if persistPath == "" {
+		panicLog.Error("persist path is empty, aborting panic report generation")
+		return
+	}
+	tl := os.Getenv("LOTUS_PANIC_JOURNAL_LOOKBACK")
+	if tl != "" && PanicReportJournalTail == defaultJournalTail {
+		i, err := strconv.Atoi(tl)
+		if err == nil {
+			PanicReportJournalTail = i
+		}
+	}
+
+	reportPath := filepath.Join(persistPath, PanicReportingPath, generateReportName(label))
```

Comment:
Can we drop `PanicReportingPath` from here, and just use persistPath - then when we call this function we'd:
* Use --panic-reports if set
* Use `Join(repoPath, PanicReportingPath)` otherwise